package/network/services/lldpd/files/lldpd.init

```diff
@@ -48,75 +115,239 @@ write_lldpd_conf()
+	fi
 	config_get lldp_class 'config' 'lldp_class'
 	config_get lldp_location 'config' 'lldp_location'
+	config_get_bool lldp_no_version 'config' 'lldp_no_version' 0
+	if [ "$CONFIG_LLDPD_WITH_LLDPMED" == "y" ]; then
+		config_get_bool lldpmed_no_inventory 'config' 'lldpmed_no_inventory' 0
+	fi
 	config_get_bool readonly_mode 'config' 'readonly_mode' 0
-	config_get agentxsocket 'config' 'agentxsocket'
+	if [ "$CONFIG_LLDPD_WITH_SNMP" == "y" ]; then
+		config_get agentxsocket 'config' 'agentxsocket'
+	fi
+	config_get filter 'config' 'filter' 15
 
-	mkdir -p /var/run/lldp
-	chown lldp:lldp /var/run/lldp
+	mkdir -p ${LLDPD_RUN}
+	chown lldp:lldp ${LLDPD_RUN}
 
 	# When lldpd starts, it also loads up what we write in this config file
 	write_lldpd_conf
 
 	procd_open_instance
-	procd_set_param command /usr/sbin/lldpd -d
+	procd_set_param command ${LLDPDBIN}
+	procd_append_param command -d
+
+	if [ $enable_lldp -gt 0 ]; then
+		if [ $force_lldp -gt 0 ]; then
+			procd_append_param command '-l'
+		fi
+	else
+		# Disable LLDP
+		procd_append_param command '-ll'
+	fi
+
+	if [ "$CONFIG_LLDPD_WITH_CDP" == "y" ] && [ $enable_cdp -gt 0 ]; then
+		if [ $cdp_version == "cdpv2" ]; then
+			if [ $force_cdp -gt 0 ]; then
+				# CDPv1 disabled, CDPv2 forced
+				procd_append_param command '-ccccc'
+			else
+				# CDPv1 disabled, CDPv2 enabled
+				procd_append_param command '-cccc'
+			fi
+		elif [ $cdp_version == "cdpv1v2" ]; then
+			if [ $force_cdp -gt 0 ] && [ $force_cdpv2 -gt 0 ]; then
+				# CDPv1 enabled, CDPv2 forced
+				procd_append_param command '-ccc'
+			elif [ $force_cdp -gt 0 ]; then
+				# CDPv1 forced, CDPv2 enabled
+				procd_append_param command '-cc'
+			else
+				# CDPv1 and CDPv2 enabled
+				procd_append_param command '-c'
+			fi
+		fi
+	fi
+
+	if [ "$CONFIG_LLDPD_WITH_FDP" == "y" ] && [ $enable_fdp -gt 0 ]; then
+		if [ $force_fdp -gt 0 ]; then
+			# FDP enbled and forced
+			procd_append_param command '-ff'
+		else
+			# FDP enabled
+			procd_append_param command '-f'
+		fi
+	fi
+
+	if [ "$CONFIG_LLDPD_WITH_SONMP" == "y" ] && [ $enable_sonmp -gt 0 ]; then
+		if [ $force_sonmp -gt 0 ]; then
+			# SONMP enabled and forced
+			procd_append_param command '-ss'
+		else
+			# SONMP enabled
+			procd_append_param command '-s'
+		fi
+	fi
+
+	if [ "$CONFIG_LLDPD_WITH_EDP" == "y" ] && [ $enable_edp -gt 0 ]; then
+		if [ $force_edp -gt 0 ]; then
+			# EDP enbled and forced
+			procd_append_param command '-ee'
+		else
+			# EDP enbled
+			procd_append_param command '-e'
+		fi
+	fi
 
-	[ $enable_cdp -gt 0 ] && procd_append_param command '-c'
-	[ $enable_fdp -gt 0 ] && procd_append_param command '-f'
-	[ $enable_sonmp -gt 0 ] && procd_append_param command '-s'
-	[ $enable_edp -gt 0 ] && procd_append_param command '-e'
 	[ $readonly_mode -gt 0 ] && procd_append_param command '-r'
+	[ $lldp_no_version -gt 0 ] && procd_append_param commanpackage/network/services/lldpd/Makefile package/network/services/lldpd/files/lldpd.initd '-k'
```

Comment:
Late breaking, but should that be as so?:
```
	[ $lldp_no_version -gt 0 ] && procd_append_param commonpackage/network/services/lldpd/Makefile package/network/services/lldpd/files/lldpd.initd '-k'
```

In fact, is this line right at all or did this slip in from some fumble?

Reply:
Seems like it should be
```sh
	[ $lldp_no_version -gt 0 ] && procd_append_param command '-k'
```

Reply:
Yep - 100% sure with testing now. This row is a fumble and needs patching.